source/core/timed-out.ts

```diff
@@ -69,7 +71,11 @@ export default function timedOut(request: ClientRequest, delays: Delays, options
 	const {host, hostname} = options;
 
 	const timeoutHandler = (delay: number, event: string): void => {
-		request.destroy(new TimeoutError(delay, event));
+		setTimeout(() => {
```

Comment:
Needs a short code comment why it's wrapped in this.

`queueMicrotask` may be a better choice here.